web/wp-content/themes/dctx-v2/inc/queries.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * Custom queries.
+ *
+ * This file is used to house "getter" function, that fetch data from the database.
+ *
+ * @package dctx
+ */
+
+/**
+ * Build the query arguments for recent posts.
+ *
+ * Depending on the categories and tags a user selects via ACF, build the query args.
+ *
+ * @param bool|array $categories Categories selected from ACF.
+ * @param bool|array $tags Tags selected from ACF.
+ * @return array $args The query arguments.
+ * @author Eric Fuller, Jeffrey de Wit
+ */
+function dctx_get_recent_posts_query_arguments( $categories, $tags ) {
+
+	$args = array();
+
+	// If no tags and just categories.
+	if ( ! $tags && $categories ) {
+		$args['category__in'] = $categories;
+	}
+
+	// If no categories and just tags.
+	if ( ! $categories && $tags ) {
+		$args['tag__in'] = $tags;
+	}
+
+	// If both categories and tags.
+	if ( $categories && $tags ) {
+		$args = array_merge(
+			$args,
+			array(
+				'tax_query' => array(
```

Comment:
:warning: **Warning**: Detected usage of tax_query, possible slow query (*WordPress.DB.SlowDBQuery.slow_db_query_tax_query*).